webapp/src/main/webapp/src/app/sender/archive/sender-archive.component.ts

```diff
@@ -0,0 +1,91 @@
+import { Component, OnInit } from '@angular/core';
+import { select, Store } from '@ngrx/store';
+import { Subscription } from 'rxjs';
+import { BaseListingComponent } from 'src/app/shared/base-listing.component';
+import { SearchEvent } from './search-event';
+import { SenderSyncArchive } from './sender-archive';
+import { SenderArchiveService } from './sender-archive.service';
+import { SenderArchivedLoaded } from './state/sender-archive.actions';
+import { GET_SYNC_ARCHIVE } from './state/sender-archive.reducer';
+
+@Component({
+	selector: 'app-sender-archive',
+	templateUrl: './sender-archive.component.html',
+	styleUrls: ['./sender-archive.component.scss']
+})
+export class SenderArchiveComponent extends BaseListingComponent implements OnInit {
+
+	count?: number;
+
+	loadedSubscription?: Subscription;
+
+	senderArchiveItens?: SenderSyncArchive[];
+
+	searchEvent: SearchEvent = new SearchEvent;
```

Comment:
This component is not correctly implementing the redux pattern

In theory you could implement this in 2 ways, by reusing the existing loaded action or you can decide to get a little fancier with a new extra action which I think should be the correct thing todo.

With the first option, the click event handler should call searchByPeriod, which should emit a loaded event so that the component refreshes its UI, this option a downside that if something goes wrong, the UI keeps showing the old items.

The second approach which is the correct way is to add new action to trigger the search, i.e. the click handler emits the new search action, the reducer for this action clears the state so that the component immediately refreshes to clears the table contents in the UI, when the search results come back you then emit the loaded events and the component refreshes to displays the search results, with this approach in the event something goes wrong the UI is blank with nothing to display. Are you familiar with NG effects? It is useful when you have to deal with things http like errors etc. but we don't have to use it now.

Reply:
I have tried to minimize direct change to the data on searchByPeriod method. You may take a look at that.

Reply:
You didn't implement what I said in my comment, please re-read it carefully

Reply:
I understand your point of view but the community always recommends using effects in case of http calls. Can we follow this approach?

Reply:
I understand your point of view but the community always recommends using effects in case of http calls in order to avoid calling services in the reducer function.

Reply:
In fact you don't a new action, just emit SENDER_ARCHIVED_LOADED with an empty map of count and items